README.md

```diff
@@ -133,65 +236,110 @@ Available architectures:
- --install           Install build dependencies.
- --arch=<arch>       Set comma-separated list of target architectures.
- --help,-h           Print this message.
+USAGE: 
+-a, --arch=<name>       run build for <name> architecture
+-d, --docker=<name>     run build in docker (name=docker environment)
+-h, --help              print this help text
+-i, --install           install dependencies
+-e, --print-env=[what]  print environment setup: packages, meta, deps
+                        (deps packages)
+-l, --print-list        print list: env (list of available environments)
+-c, --container=[opts]  use one container per kind instead of container
+                        per package; opts can be (comma separated):
+                        - clean - create new container before first use
+-r, --recreate          forcibly recreate docker images (implies -c=clean)
+
+Build elements:
+-b, --build             run build stage
+-s, --source            run source stage
+-t, --tar               run tarball creation stage
+-p, --deps              run dependency compilation stage
 
-If none of '--deps', '--source', '--build' or '--tar' options are specified,
-all of them are activated.
+When no -b, -s, -t or -p are set, all are enabled. Setting one of them will
+disable unset parameters (eg. setting -b -s will run only source and build
+stage).
 
+If --docker is set, build process will run in new container. By adding --container
+scripts will use one container to build all packages (instead of container
+per package)
 ```
 
 This script  will run build of  each package in new  docker container installing
 all build dependencies each time (with `--docker` option), for example:
 ```
-./build-env/bin/pelion-build-all.sh --docker=focal --arch=amd64,armhf,arm64
+./build-env/bin/build-all.sh --docker=focal --arch=amd64,armhf,arm64
 ```
 
+By adding `--container`  option, one container will be used  if possible. Adding
+`--container clean` will remove existing container before build.
+
 It  is  possible to  manually  run  docker and  then  build  everything in  this
 container:
 ```
-$ ./build-env/bin/docker-run.sh pelion-bionic-source
-user@95a30883d637:/pelion-build$ ./build-env/bin/pelion-build-all.sh --install --arch=amd64
+$ ./build-env/bin/docker-run-env.sh bionic
+user@95a30883d637:/pelion-build$ ./build-env/bin/build-all.sh --install --arch=amd64
+```
+
+When  build  requires  different  architecture (when  cross-compilation  is  not
+available,  like for  *Red  Hat*)  `--arch=<architecture>` has  to  be added  to
+`docker-run-env.sh` before specifying target environment:
+```bash
+./build-env/bin/docker-run-env.sh --arch arm64 rhel
 ```
 
-The option `--install` is required when the script is executed manually in clear
-docker container, otherwise it will fail due to missing build dependencies.
+The option `--install`  is required when the script is  executed in clear docker
+container, otherwise it will fail due to missing build dependencies.
```

Comment:
What does "clear" docker container mean here?

Reply:
I can change it to *new docker container* - with `-c` option we can reuse container, so "clear" means container which were not used before.

Reply:
Is this clearer?
> The option `--install`  is required during a clean build, otherwise it will fail due to missing build dependencies.

Reply:
> Is this a more obvious way to say it?
> 
> > The option `--install`  is required during a clean build, otherwise it will fail due to missing build dependencies.

I think this may be confused with content of `build` directory. We can have clean container but not clean build directory.
So re-running build with `-c=clean` will require `--install`, but re-running build with `-c` - will not.

Reply:
I agree this is confusing :)